core/trino-spi/src/main/java/io/trino/spi/type/SqlTimeWithTimeZone.java

```diff
@@ -35,6 +37,20 @@
 
     public static SqlTimeWithTimeZone newInstance(int precision, long picoseconds, int offsetMinutes)
     {
+        if (precision < 0 || precision > 12) {
```

Comment:
why not in constructor.
If we happen to add new helper factory method we may forget about those.

Reply:
i was keeping this in line with `SqlTime`. see other `SqlTime*` classes

Reply:
Yeah - I saw that later.